### PR TITLE
Add limit and offset support

### PR DESCRIPTION
**Pull request by armincerf**
Want to do order-by but its hard so leaving it for now

### CODE REVIEW

File: src/juxt/site/alpha/graphql.clj [hunk 1]
```diff
@@ -68,11 +92,9 @@
     (first results)))
 
 (defn infer-query
-  [db type]
-  (let [default-query '{:find [e]
-                        :in [type]
-                        :where [[e :juxt.site.alpha/type type]]}
-        results (for [[e] (xt/q db default-query type)]
+  [db type query]
+  (prn "Infering query for " type)
```
Comment (malcolmsparks) [inline src/juxt/site/alpha/graphql.clj:96]: Could you remove this post-merge? @armincerf

**Review by malcolmsparks — APPROVED** [review]

**Comment by armincerf** [comment]
I've added a few more features, split mostly into separate commits. Would like to get the 'this follows the graphql spec' seal of approval before merging because I'm doing some things that I'm not sure about, especially around the mutations

File: src/juxt/site/alpha/graphql.clj [hunk 2]
```diff
@@ -8,20 +8,31 @@
...
    [juxt.grab.alpha.parser :as parser]
+   [juxt.site.alpha.util :refer [assoc-some]]
    [clojure.string :as str]
+   [clojure.set :refer [rename-keys]]
    [crux.api :as xt]
    [clojure.tools.logging :as log]
    [clojure.walk :refer [postwalk]]
-   [clojure.edn :as edn]))
+   [clojure.edn :as edn]
+   [tick.core :as t]))
```
Comment (reborg) [inline src/juxt/site/alpha/graphql.clj:18]: Should this have the related dependency in deps.edn?

**Comment by malcolmsparks** [comment]
I'll definitely want a separate PR for the XT upgrade :)

**Pull request closed without merging**